packages/blockchain-wallet-v4-frontend/src/data/components/buySell/sagas.ts

```diff
@@ -1011,7 +1011,12 @@ export default ({ api, coreSagas, networks }: { api: APIType; coreSagas: any; ne
     const { isFlow, method } = payload
     const cryptoCurrency = S.getCryptoCurrency(yield select()) || 'BTC'
     const originalFiatCurrency = S.getFiatCurrency(yield select())
-    const fiatCurrency = method.currency || S.getFiatCurrency(yield select())
+    // At this point fiatCurrency should be set inside buy/sell flow - fallback to USD
+    let fiatCurrency = S.getFiatCurrency(yield select()) || 'USD'
```

Comment:
Could just use `originalFiatCurrency` here?

Reply:
at the minimum i assume this could be `originalFiatCurrency || 'USD'`